src/video/wayland/SDL_waylandopengles.c

```diff
@@ -125,7 +125,15 @@ int Wayland_GLES_SwapWindow(_THIS, SDL_Window *window)
         struct wl_display *display = videodata->display;
         SDL_VideoDisplay *sdldisplay = SDL_GetDisplayForWindow(window);
         /* ~10 frames (or 1 sec), so we'll progress even if throttled to zero. */
-        const Uint64 max_wait = SDL_GetTicksNS() + (sdldisplay->current_mode.refresh_rate ? ((SDL_NS_PER_SECOND * 10) / sdldisplay->current_mode.refresh_rate) : SDL_NS_PER_SECOND);
+        Uint64 max_wait;
+
+        if (sdldisplay->current_mode.refresh_rate_denominator && sdldisplay->current_mode.refresh_rate_numerator) {
+            int rr = sdldisplay->current_mode.refresh_rate_numerator / sdldisplay->current_mode.refresh_rate_denominator;
```

Comment:
Integer division here won't have the same rounding as before

Reply:
should be fixed with double